Tests/NIOTests/ChannelTests.swift

```diff
@@ -1717,4 +1717,386 @@ public class ChannelTests: XCTestCase {
+                }
+
+                if let connectPromise = self.connectPromise {
+                    XCTAssertTrue(connectPromise.futureResult.fulfilled, "connectPromise not fullfilled before channelInactive was called")
+                } else {
+                    XCTFail("connect(...) not called before channelInactive")
+                }
+
+                if let closePromise = self.closePromise {
+                    XCTAssertTrue(closePromise.futureResult.fulfilled, "closePromise not fullfilled before channelInactive was called")
+                } else {
+                    XCTFail("close(...) not called before channelInactive")
+                }
+
+                XCTAssertFalse(ctx.channel.closeFuture.fulfilled)
+            }
+
+            public func channelRegistered(ctx: ChannelHandlerContext) {
+                XCTAssertFalse(ctx.channel.isActive)
+                XCTAssertNil(self.connectPromise)
+                XCTAssertNil(self.closePromise)
+
+                if let registerPromise = self.registerPromise {
+                    XCTAssertTrue(registerPromise.futureResult.fulfilled, "registerPromise not fullfilled before channelRegistered was called")
+                } else {
+                    XCTFail("register(...) not called before channelRegistered")
+                }
+
+                XCTAssertFalse(ctx.channel.closeFuture.fulfilled)
+            }
+
+            public func channelUnregistered(ctx: ChannelHandlerContext) {
+                XCTAssertFalse(ctx.channel.isActive)
+
+                if let registerPromise = self.registerPromise {
+                    XCTAssertTrue(registerPromise.futureResult.fulfilled, "registerPromise not fullfilled before channelUnregistered was called")
+                } else {
+                    XCTFail("register(...) not called before channelUnregistered")
+                }
+
+                if let connectPromise = self.connectPromise {
+                    XCTAssertTrue(connectPromise.futureResult.fulfilled, "connectPromise not fullfilled before channelUnregistered was called")
+                } else {
+                    XCTFail("connect(...) not called before channelUnregistered")
+                }
+
+                if let closePromise = self.closePromise {
+                    XCTAssertTrue(closePromise.futureResult.fulfilled, "closePromise not fullfilled before channelUnregistered was called")
+                } else {
+                    XCTFail("close(...) not called before channelUnregistered")
+                }
+
+                XCTAssertFalse(ctx.channel.closeFuture.fulfilled)
+            }
+
+            public func register(ctx: ChannelHandlerContext, promise: EventLoopPromise<Void>?) {
+                XCTAssertNil(self.registerPromise)
+                XCTAssertNil(self.connectPromise)
+                XCTAssertNil(self.closePromise)
+
+                promise?.futureResult.whenSuccess {
```

Comment:
Do you actually accept this promise being nil?

Reply:
nope it should not be null ever so I think we can just make it a force unwrap.